src/CMakeLists.txt

```
@@ -25,6 +25,10 @@ if( CMAKE_BUILD_TYPE STREQUAL "Debug" )
     target_compile_definitions( T8 PUBLIC T8_ENABLE_DEBUG )
 endif()
 
+if( T8CODE_EXPORT_COMPILE_COMMANDS )
+  target_compile_definitions(T8 PRIVATE CMAKE_EXPORT_COMPILE_COMMANDS=1)
```

```suggestion
  set_target_properties( T8 PROPERTIES EXPORT_COMPILE_COMMANDS ON )
```
The `target_compile_definitions` did not work on my side, however this seem to work. I would also suggest adding the same property to every test/benchmark as well or advise users in the wiki to globally set the variable `CMAKE_EXPORT_COMPILE_COMMANDS` in the build command (e.g. `cmake path/to/t8code -DCMAKE_EXPORT_COMPILE_COMMANDS=ON`) to export compile commands for all targets automatically.

---

Done :)